apps/src/templates/levelSummary/SummaryResponses.jsx

```diff
@@ -91,6 +98,15 @@ const SummaryResponses = ({
   // only when the policy allows it for that user.
   const showAnswerToggle = scriptData.answer_is_visible && isMulti;
 
+  const toggleNames = () => {
+    if (showStudentNames) {
+      logEvent(EVENTS.CFU_NAMES_TOGGLED_OFF);
+    } else {
+      logEvent(EVENTS.CFU_NAMES_TOGGLED_ON);
+    }
+    setShowStudentNames(!showStudentNames);
```

Comment:
```suggestion
    setShowStudentNames((prevShowStudentNames => !prevShowStudentNames);
```
This prevents a race condition from spamming the toggle button back and forth.

We may also want to [debounce](https://www.geeksforgeeks.org/lodash-_-debounce-method/) the logging so it doesn't get spammed, but I think that's probably not too important/necessary.